test/rpc/test_rpc_schema.py

```diff
@@ -0,0 +1,320 @@
+                "entry_point_selector": ["0x01", "0x02"],
+                "calldata": [],
+            },
+            "block_id": "latest",
+        },
+    ),
+)
+def test_schema_raises_on_invalid_kwargs(params):
+    """
+    Call params validation with kwargs
+    """
+    resp = rpc_call("starknet_call", params=params)
+
+    assert "error" in resp
+    error = resp["error"]
+
+    assert error["code"] == RpcErrorCode.INVALID_PARAMS.value
+
+
+@pytest.mark.usefixtures("run_devnet_in_background")
+@pytest.mark.parametrize(
+    "params",
+    (
+        [
+            {
+                "contract_address": "0x01",
+                "entry_point_selector": rpc_felt(get_selector_from_name("get_balance")),
+                "calldata": [],
+            }
+        ],
+        [
+            "latest",
+        ],
+        [
+            {
+                "contract_address": "0x01",
+                "entry_point_selector": rpc_felt(get_selector_from_name("get_balance")),
+                "calldata": [],
+            },
+            12345,
+        ],
+        [
+            {
+                "contract_address": "0x01",
+                "entry_point_selector": rpc_felt(get_selector_from_name("get_balance")),
+                "calldata": [],
+            },
+            "aaaeee",
+        ],
+        [],
+        [
+            {
+                "entry_point_selector": rpc_felt(get_selector_from_name("get_balance")),
+                "calldata": [],
+            },
+            "latest",
+        ],
+        [
+            {
+                "contract_address": "0x01",
+                "entry_point_selector": rpc_felt(get_selector_from_name("get_balance")),
+                "calldata": "0x1234",
+            },
+            "latest",
+        ],
+        [
+            {
+                "contract_address": 1324,
+                "entry_point_selector": rpc_felt(get_selector_from_name("get_balance")),
+                "calldata": [],
+            },
+            "latest",
+        ],
+        [
+            {
+                "contract_address": "0x01",
+                "entry_point_selector": ["0x01", "0x02"],
+                "calldata": [],
+            },
+            "latest",
+        ],
+        [
+            "latest",
+            {
+                "contract_address": "0x01",
+                "entry_point_selector": "0x01",
+                "calldata": [],
+            },
+        ],
+    ),
+)
+def test_schema_raises_on_invalid_args(params):
+    """
+    Test params validation with args
+    """
+    resp = rpc_call("starknet_call", params=params)
+
+    assert "error" in resp
+    error = resp["error"]
+
+    assert error["code"] == RpcErrorCode.INVALID_PARAMS.value
+
+
+@pytest.mark.usefixtures("run_devnet_in_background")
+@pytest.mark.parametrize(
+    "run_devnet_in_background",
+    [["--disable-rpc-request-validation"]],
+    indirect=True,
+)
+def test_schema_does_not_raise_on_disabled_request_validation():
```

Comment:
I wanted to also add a test for `--disable-rpc-response-validation` but I don't know how to approach that: It'd require somehow mocking the route so it returns incorrect value. 

I did some manual testing and it seems to be working.

Reply:
First I'd like to hear why we need that flag (asked that in another PR comment)